src/main/groovy/com/github/jengelman/gradle/plugins/shadow/relocation/SimpleRelocator.groovy

```diff
@@ -188,4 +190,39 @@ class SimpleRelocator implements Relocator {
             return sourceContent.replaceAll("\\b" + pattern, shadedPattern)
         }
     }
+
+    @Input
+    String getPattern() {
+        return pattern
+    }
+
+    @Input
+    String getPathPattern() {
+        return pathPattern
+    }
+
+    @Input
+    String getShadedPattern() {
+        return shadedPattern
+    }
+
+    @Input
+    String getShadedPathPattern() {
+        return shadedPathPattern
+    }
```

Comment:
@ghale

These properties are immutable and unused, can we remove them? I'm planning to remove them and migrate `includes` and `excludes` to `org.gradle.api.provider.SetProperty`.

Reply:
Yeah, if these are being removed in favor of property objects, then yes, these can be removed.  They were only added in order to expose the inputs from the relocator.

Reply:
To be clear, the new property objects should also be exposed as inputs (i.e. a getter with an `@Input` annotation) in order to maintain proper cacheability after these are removed.

Reply:
Can you take a review for #1079?